### PR TITLE
stable 2.1 | adapt to new cri containerd cni tarball link

### PR DESCRIPTION
**Pull request by snir911**
continuing https://github.com/kata-containers/tests/pull/3713 work

### CODE REVIEW

**Comment by snir911** [comment]
@lifupan, @fidencio CI's previous error seems to pass now, however, there's another failure

```
14:21:40 
14:21:40     failed to remove container: rpc error: code = Unknown desc = failed to set removing state for container "a62d6b93cd87513b6c51a31e25298c6bbee21bacd41f4be7f162633dd9feb6da": container is still running, to stop first
14:21:40     Unexpected error:
14:21:40         <*status.statusError | 0xc00030a5f0>: {
14:21:40             Code: 2,
14:21:40             Message: "failed to set removing state for container \"a62d6b93cd87513b6c51a31e25298c6bbee21bacd41f4be7f162633dd9feb6da\": container is still running, to stop first",
14:21:40             Details: nil,
14:21:40             XXX_NoUnkeyedLiteral: {},
14:21:40             XXX_unrecognized: nil,
14:21:40             XXX_sizecache: 0,
14:21:40         }
14:21:40         rpc error: code = Unknown desc = failed to set removing state for container "a62d6b93cd87513b6c51a31e25298c6bbee21bacd41f4be7f162633dd9feb6da": container is still running, to stop first
14:21:40     occurred
```
which IIUC related to https://github.com/containerd/containerd/pull/4656 that is fixed only in containerd 1.5+

**Comment by fidencio** [comment]
> @lifupan, @fidencio CI's previous error seems to pass now, however, there's another failure
> 
> ```
> 14:21:40 
> 14:21:40     failed to remove container: rpc error: code = Unknown desc = failed to set removing state for container "a62d6b93cd87513b6c51a31e25298c6bbee21bacd41f4be7f162633dd9feb6da": container is still running, to stop first
> 14:21:40     Unexpected error:
> 14:21:40         <*status.statusError | 0xc00030a5f0>: {
> 14:21:40             Code: 2,
> 14:21:40             Message: "failed to set removing state for container \"a62d6b93cd87513b6c51a31e25298c6bbee21bacd41f4be7f162633dd9feb6da\": container is still running, to stop first",
> 14:21:40             Details: nil,
> 14:21:40             XXX_NoUnkeyedLiteral: {},
> 14:21:40             XXX_unrecognized: nil,
> 14:21:40             XXX_sizecache: 0,
> 14:21:40         }
> 14:21:40         rpc error: code = Unknown desc = failed to set removing state for container "a62d6b93cd87513b6c51a31e25298c6bbee21bacd41f4be7f162633dd9feb6da": container is still running, to stop first
> 14:21:40     occurred
> ```
> 
> which IIUC related to [containerd/containerd#4656](https://github.com/containerd/containerd/pull/4656) that is fixed only in containerd 1.4+ would it be to intrusive to bump to 1.4 in stable?

I think it would, but if this is what takes to have it out, then this is what takes to have it out.
We have to have a huge release note saying we stop supporting containerd 1.3.x after we do that.

@egernst, @mxpv opinions?

**Comment by jodh-intel** [comment]
Thanks @snir911. Could you state in the commit that this is a backport of https://github.com/kata-containers/tests/pull/3629? You'll also need to add the `backport` label.

File: integration/containerd/cri/integration-tests.sh [hunk 1]
```diff
@@ -327,9 +332,18 @@ main() {
 	# make sure cri-containerd test install the proper critest version its testing
 	rm -f "${CRITEST}"
 
-	pushd "${GOPATH}/src/${cri_containerd_repo}"
+	go get ${cri_repo}
+	pushd "${GOPATH}/src/${cri_repo}"
+
+	git reset HEAD
+	git checkout master
```
Comment (lifupan) [inline integration/containerd/cri/integration-tests.sh:339]: here you shouldn't use the master branch, instead you should use the 1.3.10 corresponding cri branch.

**Comment by lifupan** [comment]
Hi @snir911  @fidencio 

I had found the root cause, the containerd cri's commit https://github.com/containerd/containerd/pull/4362/commits/d7e59b141cfc608e89b80693862064daac19c28e enabled the force remove container, even the container was running, and the cri-tools's PR https://github.com/kubernetes-sigs/cri-tools/pull/633 added the corresponding test cases. Thus we should keep the cri-tools's version compatible with the containerd/cri. But this PR switched the containerd to 1.3.10 which hadn't support the force remove container, but you used the cri-tools master branch which had enabled the force remove container test case, which caused the test failed. So you should switch the containerd/cri to the corresponding version to fix this issue.

**Comment by snir911** [comment]
/retest-ubuntu
Thanks @lifupan!  pushed, lets see

File: integration/containerd/cri/integration-tests.sh [hunk 2]
```diff
@@ -327,9 +332,18 @@ main() {
...
 	rm -f "${CRITEST}"
 
-	pushd "${GOPATH}/src/${cri_containerd_repo}"
+	go get ${cri_repo}
+	pushd "${GOPATH}/src/${cri_repo}"
+
+	git reset HEAD
+	git checkout release/1.3
+	# switch to the default pause image set by containerd:1.5.x
+	sed -i 's#k8s.gcr.io/pause:3.[0-9]#k8s.gcr.io/pause:3.5#' integration/main_test.go
```
Comment (lifupan) [inline integration/containerd/cri/integration-tests.sh:341]: This change should also be dropped, since it only work for 1.5.x

**Comment by fidencio** [comment]
@snir911, @lifupan, thanks for working on this.

Seems that things are passing now, and we have a bunch of dependent PRs.
@snir911, would be possible to gather all the PRs that have to be backported in order to get this one working and put together with https://github.com/kata-containers/kata-containers/pull/2226?

**Comment by snir911** [comment]
@fidencio this is kata-containers/tests repo :) (tests backports were already mereged)

**Comment by fidencio** [comment]
You're right. Let me get https://github.com/kata-containers/kata-containers/pull/2222 merged and we can move on with this one.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @snir911!
Also, thanks @lifupan!

**Review by cmaf — APPROVED** [review]
LGTM

**Pull request merged**